### PR TITLE
Update: add `ignoreComments` option to `indent` rule (fixes #9018)

### LINKED ISSUE #9018: Ignore comments option for indent rule
**What rule do you want to change?**

`indent`

**Does this change cause the rule to produce more or fewer warnings?**

Potentially fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option:

```json
{
    "indent": ["error", 4, { "comments": "off" }]
}
```

This option will completely disregard lines beginning with comments and allow them to be at any indentation level.

**Please provide some example code that this change will affect:**

```js
doSomething();  // This comment is fine
                           // This comment is reported, but shouldn't be

if (foo) {
    doBar();

// This comment is reported, but shouldn't be with new option
// (maybe comment should be easy to spot via outdenting?)
    doBaz();
}
```

**What does the rule currently do for this code?**

Reports all comments (except trailing comment on first line)

**What will the rule do after it's changed?**

Reports none of the comments

### PR DESCRIPTION
**Pull request by platinumazure**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

See #9018.

**What changes did you make? (Give an overview)**

Added `ignoreComments` option (default: false) to indent rule.

**Is there anything you'd like reviewers to focus on?**

This seemed, um, simpler than I thought. Is there a more idiomatic way I should have done this? Any advice on how to test performance and make sure it's acceptable?

### CODE REVIEW

File: docs/rules/indent.md [hunk 1]
```diff
@@ -84,7 +84,8 @@ This rule has an object option:
...
 * `"flatTernaryExpressions": true` (`false` by default) requires no indentation for ternary expressions which are nested in other ternary expressions.
-* `ignoredNodes` accepts an array of [selectors](/docs/developer-guide/selectors.md). If an AST node is matched by any of the selectors, the indentation of tokens which are direct children of that node will be ignored. This can be used as an escape hatch to relax the rule if you disagree with the indentation that it enforces for a particular syntactic pattern.
+* `"ignoredNodes"` accepts an array of [selectors](/docs/developer-guide/selectors.md). If an AST node is matched by any of the selectors, the indentation of tokens which are direct children of that node will be ignored. This can be used as an escape hatch to relax the rule if you disagree with the indentation that it enforces for a particular syntactic pattern.
+* `"ignoreComments"` (default: false) can be used when comments do not need to be aligned with nodes on the previous or next line.
```
Comment (not-an-aardvark) [inline docs/rules/indent.md]: I'm wondering if we should document the existing comment behavior about allowing alignment with the previous/next line. It seems like this description hints at the idea, but I'm not sure if it's explained more thoroughly anywhere.

**Comment by gyandeeps** [comment]
what about block comments?

```js
/* block-line comment */
    var a = 1;


/*
block comment
*/
    var a = 1;
```

**Comment by platinumazure** [comment]
@gyandeeps I think this change should work for block comments too (and I think the rule currently does not check for alignment of text in later lines of the block comment, but I'll check). I'll add tests for block comments.

@not-an-aardvark Thanks, I'll do some performance tests after I make the suggested changes.

**Comment by platinumazure** [comment]
Apologies for not leaving an update sooner. I've been doing Christmas shopping last-minute :laughing: 

# Block comments

I added basic tests for just checking the alignment of start of block comment. @gyandeeps Given you approved the PR, hopefully that looks good to you.

However, I wasn't sure if you asking about indentation/alignment of consecutive lines in a comment. If so, I haven't tested that. At the time, I was thinking [`multiline-comment-style`](https://eslint.org/docs/rules/multiline-comment-style) should be responsible for those, but please let me know if you think differently.

That said, I don't think indent currently handles those continuing lines in block comments, so I don't particularly want to add a lot of extra logic in this PR.

# Performance

I didn't capture the metrics, but I did follow @not-an-aardvark's suggestions and saw that performance was unchanged. I even tried changing our lint settings to use `ignoreComments: true` (and otherwise identical settings) and saw no significant change in performance.

Let me know if I should run any more tests or show some metrics.

File: docs/rules/indent.md (hunk 1 above)
Comment (platinumazure) [reply docs/rules/indent.md]: The word "comment" isn't even mentioned in the docs currently, so I would hazard a guess that there is no description of expected comment indentation. If you want me to add that to this pull request, feel free to Request Changes here.

Comment (not-an-aardvark) [reply docs/rules/indent.md]: ¯\\\_(ツ)_/¯ It's not a big problem, I'm fine with this being merged as-is.

**Comment by platinumazure** [comment]
@eslint/eslint-team Anyone else want to review this?

**Pull request merged**